src/UI/Component/ViewControl/Mode.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\ViewControl;
+
+/**
+ * This describes a Mode Control
+ */
+interface Mode extends \ILIAS\UI\Component\Component {
+	/**
+	 * set the currently active Button by label.
+	 *
+	 * @return string the label of the button to activate
+	 */
+	public function withActive();
```

Comment:
I also think this could go away when we enforce the "one button is active" constraint through the factory method.